Closes #110

Arne2 opened the pull request:
## Beschreibung:

Bitte nochmal drüberschaun, gerade ob ich irgend welche teile generischer gestalten kann oder eher effizenten code verwenden kann.
## Branch-Checklist:
- [ ] MicroService getestet
- [ ] GUI getestet
- [ ] Barakuda Issue erstellt
## Bestätigungen:
- [ ] @darenegade 
- [x] @peter-mueller 
- [x] @FabianHoltkoetter
## Referenz:

closes #110

Arne2 committed:
#110 grid added, all methods not yet implemented

Arne2 committed:
Merge branch 'master' into as-#110

Arne2 committed:
#110 zwischencommit

Arne2 committed:
#110 funktionen aktualisiert - almost closed

Arne2 committed:
#110 closed

darenegade commented:
Bitte nicht die GenericTable als BeanContainer missbrauchen ;)

Das Grid sollte wie die GenericTable eine eigene Klasse (Component) sein und generisch aufgebaut werden, je nachdem welche Entität dargestellt werden soll. Hier kannst du dich an der GenericTable orientieren, wie diese die Spalten einer Entität ermittelt (@Fieldidentifier).
In der View wird dann nur noch die GenericGrid eingebunden.

Arne2 committed:
#110 Grid wird generisch gemacht - Stand Freitag

Arne2 committed:
#110 closed - Grid jetzt GenericGrid und es exestiert für die BuergerTableView eine BuergerGrid
Wichtig: wenn kinder löschen bzw. Partner löschen implementiert wird, ggf. über die ClickListener schaun, ob die richtig implementiert sind

Arne2 committed:
#110 closed - Grid jetzt GenericGrid und es exestiert für die BuergerTableView eine BuergerGrid
Wichtig: wenn kinder löschen bzw. Partner löschen implementiert wird, ggf. über die ClickListener schaun, ob die richtig implementiert sind

peter-mueller commented on GUI-Lib/src/main/java/de/muenchen/vaadin/guilib/components/GenericGrid.java:
```diff
@@ -0,0 +1,108 @@
+
+        this.setContainerDataSource(new BeanItemContainer<>(entityClass));
+
+        // create table
```
Warum wird die `Table table` erstellt?

peter-mueller commented on GUI-Lib/src/main/java/de/muenchen/vaadin/guilib/components/GenericGrid.java:
```diff
@@ -0,0 +1,108 @@
+     *
+     * @param entity the entity
+     */
+    public void add(Optional<T> entity) {
```
entity.ifPresent(getContainerDataSource()::addItem)

peter-mueller commented on GUI-Lib/src/main/java/de/muenchen/vaadin/guilib/components/GenericGrid.java:
```diff
@@ -0,0 +1,108 @@
+    public void addAll(List<T> entity) {
+        LOG.debug("added search result");
+        this.getContainerDataSource().removeAllItems();
+        entity.stream().forEach(item -> this.getContainerDataSource().addItem(item));
```
entities.forEach(getContainerdataSource()::addItem);

peter-mueller commented on GUI-Lib/src/main/java/de/muenchen/vaadin/guilib/components/GenericGrid.java:
```diff
@@ -0,0 +1,108 @@
+    @Override
+    public void accept(reactor.bus.Event<ComponentEvent<T>> eventWrapper) {
+        ComponentEvent event = eventWrapper.getData();
+
```
Wenn man equals auf dem enum (z.B. .SAVE) aufruft kann man sicher sein dass es keine NullPointerExeption gibt.

peter-mueller commented on vadin-demo-gui/src/main/java/de/muenchen/vaadin/ui/components/BuergerGrid.java:
```diff
@@ -1,69 +1,221 @@
+    private ActionButton copy;
+    private ActionButton delete;
 
     public BuergerGrid(final BuergerViewController controller) {
```
Das ist m.M. nach sehr komplex und könnte in mehrere private Methoden aufgeteilt werden

peter-mueller commented on vadin-demo-gui/src/main/java/de/muenchen/vaadin/ui/components/BuergerGrid.java:
```diff
@@ -1,69 +1,221 @@
     }
 
+    private void setButtonVisability() {
+        int size = grid.getSelectedRows().size();
```
evtl. nicht nach size sondern nach Buttons sortierten 

z.B.

``` java
edit.setVisible(size == 1);
copy.setVisible(size > 0);
delete.setVisible(size > 0);
```

FabianWilms commented on vadin-demo-gui/src/main/java/de/muenchen/vaadin/ui/components/BuergerGrid.java:
```diff
@@ -1,69 +1,221 @@
- * @author claus.straube
+ * @author claus.straube, arne.schoentag
  */
 public class BuergerGrid extends CustomComponent {
```
Warum extended BuergerGrid nicht einfach GenericGrid?

Arne2 committed:
#110 closed - Verbesserungsvorschläge angenommen und umgesetzt

FabianWilms commented:
Löschen auf GUI-Seite funktioniert noch nicht. Außerdem müssen abhängigkeiten in die POM eingetragen werden. @Arne2
